### PR TITLE
Feature/80220 added organisation details page draft

### PR DESCRIPTION
**Pull request by rafalgajewski**

### CODE REVIEW

File: src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs [hunk 1]
```diff
@@ -40,8 +40,32 @@ public async Task<IActionResult> Index()
...
+                    new ActionModel($"Manage your account", string.Empty, "Dashboard"),
                 }));
     }
+
+    [HttpGet("organization-details")]
+    public async Task<IActionResult> OrganizationDetails()
+    {
+        var organisationResult = await _mediator.Send(new GetUserOrganisationInformationQuery());
+
+        var address = new List<string>
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs:53]: Query could return Model which is fitted into view

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+@model HE.InvestmentLoans.WWW.Models.OrganisationDetailsModel;
+
+<gds-div-grid-column-full>
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:3]: Please attach screenshot with the new view.

Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:3]: Please add at least one integration tests for that page

File: src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs [hunk 3]
```diff
@@ -40,8 +40,32 @@ public async Task<IActionResult> Index()
...
                 new List<ActionModel>
                 {
-                    new($"Manage {userOrganisationResult.OrganizationBasicInformation.RegisteredCompanyName} details", string.Empty, "Dashboard"),
-                    new($"Manage your account", string.Empty, "Dashboard"),
+                    new ActionModel($"Manage {userOrganisationResult.OrganizationBasicInformation.RegisteredCompanyName} details", "OrganizationDetails", "UserOrganisation"),
+                    new ActionModel($"Manage your account", string.Empty, "Dashboard"),
                 }));
     }
+
+    [HttpGet("organization-details")]
```
Comment (Antek012) [inline src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs:48]: Organisation is already in Controller name so "details" should be enought

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            Manage @Model.OrganisationName details
+        </gds-h1>
+
+        <gds-h3 class="govuk-heading-l">
+            Organisation details
+        </gds-h3>
+
+        <gds-summary-list class="govuk-!-margin-bottom-9">
+
+            @await Html.PartialAsync("_CheckAnswersRow", new ViewDataDictionary(ViewData)
```
Comment (Antek012) [inline src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:19]: I think we have to rename this partial if we want use it this way.

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml [hunk 5]
```diff
@@ -0,0 +1,63 @@
+@model HE.InvestmentLoans.WWW.Models.OrganisationDetailsModel;
+
+<gds-div-grid-column-full>
+    <form asp-controller="Organisation" asp-action="OrganisationDetails" asp-method="post" novalidate>
+
+        <gds-back-link href="@Url.Action("Index", "UserOrganisation", new {})" text="Back" class="govuk-!-margin-bottom-8">
```
Comment (Antek012) [inline src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:6]: Consider using nameof for Controller and Method instead strings .

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:3]: org_details

File: src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs (hunk 3 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs]: I agree on this one

File: src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs [hunk 6]
```diff
@@ -40,8 +40,32 @@ public async Task<IActionResult> Index()
...
+        };
+
+        return View(
+            "OrganizationDetails",
+            new OrganisationDetailsModel(
+                organisationResult.OrganizationBasicInformation.RegisteredCompanyName,
+                organisationResult.OrganizationBasicInformation.ContactInformation.PhoneNUmber,
+                address,
+                organisationResult.OrganizationBasicInformation.CompanyRegistrationNumber,
+                "Smaple org requsgsaj"));
```
[the remark this answers is not in the document]
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs]: Already changed that

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml (hunk 4 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml]: I've created new simplified partial called simpleRow

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            })
+
+            @await Html.PartialAsync("_SimpleRow", new ViewDataDictionary(ViewData)
+            {
+            { "Title", "Companies House Registration number" },
+            { "Value", Model.CompaniesHouseNumber },
+            })
+
+        </gds-summary-list>
+        @if (Model.CurrentOrganisationDataState == "You requested")
```
Comment (Carq) [inline src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:44]: Please move all possible values of Request it to const or enum.

File: src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs (hunk 1 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Controllers/UserOrganisationController.cs]: I've added OrganisationDetailsViewModel which is now returned from query

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml (hunk 2 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml:14]: Done

File: src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml (hunk 7 above)
Comment (rafalgajewski) [reply src/HE.InvestmentLoans.WWW/Views/UserOrganisation/OrganizationDetails.cshtml]: Enum added (OrganisationChangeRequestState)

**Review by burzynskijakub — APPROVED** [review]

**Review by Carq — APPROVED** [review]

**Pull request merged**